feat: added schema validation using schema utils

anikethsaha opened the pull request:
<!--
  HOLY CRAP a Pull Request. We ❤️ those!

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  Please place an x (no spaces!) in all [ ] that apply
-->

This PR contains a:

- [ ] **bugfix**
- [X] new **feature**
- [ ] **code refactor**
- [ ] **test update** <!-- if bug or feature is checked, this should be too -->
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case
To validate the options submitting to the loader.
Took default options from [less-node](https://github.com/less/less.js/blob/0fac98e24c7c2c8eee47edac95bf3731869110aa/lib/less/default-options.js)

<!--
  Please explain the motivation or use-case for your change.
  What existing problem does the PR solve?
  If this PR addresses an issue, please link to the issue.
-->

### Breaking Changes
Not really unless `"additionalProperties": true,` is false

<!--
  If this PR introduces a breaking change, please describe the impact and a
  migration path for existing applications.
-->

### Additional Info

@evilebottnawi this loader should accept the default options provided by the less node, I saw most of the test cases failed because of some other options.
Can you mention or locate me what are the other loader options it is taking 👍 
I will refactore accordingly


PS: waiting for https://github.com/webpack-contrib/less-loader/pull/308 to get merge so that I will rebase and update the lock files again

anikethsaha commented:
cc @evilebottnawi

evilebottnawi commented on src/options.json:
```diff
@@ -1,73 +1,109 @@
+      ]
     },
-    "modifyVars": {
+    "lessOption": {
```
`lessOptions`

anikethsaha replied:
👍

anikethsaha commented:
@evilebottnawi I guess its a breaking change !!
If this change if fine by you, I will go ahead with changing the tests

evilebottnawi commented:
@anikethsaha yes, it is breaking change, don't worry they will be part of next release, yes please fix tests

anikethsaha commented:
> @anikethsaha yes, it is breaking change, don't worry they will be part of next release, yes please fix tests

Sure 👍

anikethsaha commented:
Finally ready for review 

cc @evilebottnawi ,
I added tests for validate options,
removed unused dep (`memory-fs`) and added `memfs` for tests

anikethsaha commented:
Cc @evilebottnawi

evilebottnawi commented on src/options.json:
```diff
@@ -0,0 +1,103 @@
+{
+  "additionalProperties": false,
+  "properties": {
+    "lessOptions": {
```
sourceMap should be part of loader options, not Less options

No need describe all lessOptions, because in new release they will add more options

anikethsaha replied:
Less too take this option i guess. Should I keep them both? 
Not all but I am thinking of covering max options. I actually wanted to keep the `additionalProperties` to `false`

evilebottnawi replied:
For lessOptions we should set `additionalProperties`, Less can change options in any time

anikethsaha replied:
Ok then 👌

anikethsaha commented:
@evilebottnawi I moved the `sourceMap` options from `lessOptions` to loader's options. And it is being computed by the less compiler. I removed the detailed `lessOptions` property.
check this commit https://github.com/webpack-contrib/less-loader/pull/310/commits/8c379889c8f93d27739f4a00b4eddd8e0dadaedc .
if its not what you were looking for I will revert https://github.com/webpack-contrib/less-loader/pull/310/commits/8c379889c8f93d27739f4a00b4eddd8e0dadaedc

evilebottnawi commented on src/options.json:
```diff
@@ -0,0 +1,26 @@
+          "additionalProperties": true
+        },
+        {
+          "instanceof": "Function"
```
Hm, it can be `function`?

anikethsaha replied:
No, sorry my bad . will remove it

anikethsaha replied:
done 👍

anikethsaha commented:
cc @evilebottnawi CI is clean 👍

evilebottnawi commented:
Fixed in the other PR

evilebottnawi commented:
Anyway thanks for the PR
